File: LeetCode/0796/python/lc0796.py
# Runtime complexity: O(n)
# Auxiliary space complexity: O(2*n) == O(n)
# Subjective level: easy
# Solved on: 2026-02-16
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        # Answer uses the same trick as 0459/Repeated Substring Pattern.
        return len(s) == len(goal) and (s+s).find(goal) != -1

File: LeetCode/0796/golang/lc0796.go
package lc0796

import "strings"

// Runtime complexity: O(n)
// Auxiliary space: O(2n) == O(n)
// Subjective level: easy
// Solved on: 2026-05-03
func rotateString(s string, goal string) bool {
	return len(s) == len(goal) && strings.Contains(s+s, goal)
}
